pkg/server/container_create.go

```diff
@@ -154,6 +163,18 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 	if username := config.GetLinux().GetSecurityContext().GetRunAsUsername(); username != "" {
 		specOpts = append(specOpts, containerd.WithUsername(username))
 	}
+	// Set apparmor profile, (privileged or not) if apparmor is enabled
+	if AppArmorEnabled {
+		appArmorProf := config.GetLinux().GetSecurityContext().GetApparmorProfile()
+		if appArmorProf == RuntimeDefault {
+			// TODO (mikebrow): delete created apparmor default profile
+			specOpts = append(specOpts, apparmor.WithDefaultProfile(AppArmorDefaultProfileName))
+		}
+		if appArmorProf != "" {
+			// Trim default profile name prefix
+			specOpts = append(specOpts, apparmor.WithProfile(strings.TrimPrefix(appArmorProf, ProfileNamePrefix)))
+		}
+	}
```

Comment:
The behavior when profile is not specified is not clearly defined. I filed a Kubernetes issue for this: https://github.com/kubernetes/kubernetes/issues/51746

I'm fine with not setting default profile for now, until it's clearly defined. Add TODO for it.

Reply:
kk